source/MaterialXCore/Element.cpp

```diff
@@ -104,6 +104,24 @@ string Element::getNamePath(ConstElementPtr relativeTo) const
     return res;
 }
 
+ElementPtr Element::getElementByNamePath(const std::string& path)
+{
+    const std::vector<std::string> elementNames = splitString(path, NAME_PATH_SEPARATOR);
```

Comment:
Just a minor note that you can omit the "std::" qualifier for string and vector within the MaterialX namespace.

Reply:
Ah, ok, didn't realize that.